mathics/builtin/specialfns/hypergeom.py

```diff
@@ -0,0 +1,217 @@
+"""
+Hypergeometric functions
+
+See also <url>
+:Chapter 15 Hypergeometric Functions in the Digital Library of Mathematical Functions:
+https://dlmf.nist.gov/15</url>.
+"""
+
+import mpmath
+import sympy
+
+from mathics.core.attributes import (
+    A_LISTABLE,
+    A_N_HOLD_FIRST,
+    A_NUMERIC_FUNCTION,
+    A_PROTECTED,
+    A_READ_PROTECTED,
+)
+from mathics.core.builtin import MPMathFunction
+from mathics.core.convert.sympy import from_sympy
+from mathics.core.evaluation import Evaluation
+from mathics.core.number import FP_MANTISA_BINARY_DIGITS
+from mathics.eval.arithmetic import run_mpmath
+
+
+class HypergeometricPFQ(MPMathFunction):
+    """
+    <url>
+    :Generalized hypergeometric function: https://en.wikipedia.org/wiki/Generalized_hypergeometric_function</url> (<url>
+    :mpmath: https://mpmath.org/doc/current/functions/hypergeometric.html#hyper</url>, <url>
+    :sympy: https://docs.sympy.org/latest/modules/functions/special.html#sympy.functions.special.hyper.hyper</url>, <url>
+    :WMA: https://reference.wolfram.com/language/ref/HypergeometricPFQ.html</url>)
+    <dl>
+      <dt>'HypergeometricPFQ'[${a_1, ..., a_p}, {b_1, ..., b_q}, z$]
+      <dd>returns ${}_p F_q({a_1, ..., a_p}; {b_1, ..., b_q}; z)$.
+    </dl>
+    >> HypergeometricPFQ[{2}, {2}, 1]
+     = E
+
+    Result is symbollicaly simplified by default:
+    >> HypergeometricPFQ[{3}, {2}, 1]
+     = HypergeometricPFQ[{3}, {2}, 1]
+    unless a numerical evaluation is explicitly requested:
+    >> HypergeometricPFQ[{3}, {2}, 1] // N
+     = 4.07742
+
+    The following special cases are handled:
+    >> HypergeometricPFQ[{}, {}, z]
+     = 1
+    >> HypergeometricPFQ[{0}, {b}, z]
+     = 1
+    >> HypergeometricPFQ[{b}, {b}, z]
+     = E ^ z
+    """
+
+    attributes = A_NUMERIC_FUNCTION | A_PROTECTED | A_READ_PROTECTED
```

Comment:
```suggestion
    attributes = A_PROTECTED | A_READ_PROTECTED
```

WMA does not list this function as being a numeric function.